packages/inbound-email-handler/src/newsletter.ts

```diff
@@ -6,12 +6,70 @@ const EMAIL_CONFIRMATION_CODE_RECEIVED_TOPIC = 'emailConfirmationCodeReceived'
 const EMAIL_FORWARDING_SENDER_ADDRESSES = [
   'Gmail Team <forwarding-noreply@google.com>',
 ]
-const NEWSLETTER_SENDER_REGEX =
-  /<.+@((axios.com)|(mail.bloombergbusiness.com))>/
 const CONFIRMATION_CODE_PATTERN = /^\\(#\\d+\\)/
-const AXIOS_URL_PATTERN = /View in browser at <a.*>(.*)<\/a>/
-const BLOOMBERG_URL_PATTERN =
-  /<a class="view-in-browser__url" href=["']([^"']*)["']/
+
+export class NewsletterHandler {
+  protected senderRegex = /NEWSLETTER_SENDER_REGEX/
+  protected urlRegex = /NEWSLETTER_URL_REGEX/
+
+  isNewsletter(_rawUrl: string, from: string): boolean {
+    // Axios newsletter is from <xx@axios.com>
+    const re = new RegExp(this.senderRegex)
+    return re.test(from)
+  }
+
+  getNewsletterUrl(rawUrl: string, html: string): string | undefined {
+    // get newsletter url from html
+    const matches = html.match(this.urlRegex)
+    if (matches) {
+      return matches[1]
+    }
+    return undefined
+  }
+
+  async handleNewsletter(
+    email: string,
+    html: string,
+    rawUrl: string,
+    title: string,
+    from: string
+  ): Promise<[string] | undefined> {
+    console.log('handleNewsletter', email, rawUrl, title, from)
+
+    if (!email || !html || !title || !from) {
+      console.log('invalid newsletter email')
+      throw new Error('invalid newsletter email')
+    }
+
+    const url = this.getNewsletterUrl(rawUrl, html)
+    console.log('url', url)
+    if (!url) {
+      console.log('invalid newsletter url', url)
+      throw new Error('invalid newsletter url')
+    }
+
+    // get author name from email
+    // e.g. 'Jackson Harper from Omnivore App <jacksonh@substack.com>'
+    // or 'Mike Allen <mike@axios.com>'
+    const authors = from.includes(' from ')
+      ? from.split(' from')
+      : from.split(' <')
+    if (!authors) {
```

Comment:
I wonder if one of the mail parsing libraries has some methods for stuff like this.

Reply:
This one might suit our need: https://www.npmjs.com/package/addressparser

Reply:
I tried https://www.npmjs.com/package/addressparser and it seems okay